pkgs/by-name/ch/choose-gui/package.nix

```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  xcbuild,
+  coreutils,
+  darwin,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "choose-gui";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "chipsenkbeil";
+    repo = "choose";
+    rev = version;
+    hash = "sha256-oR0GgMinKcBHaZWdE7O+mdbiLKKjkweECKbi80bjW+c=";
+  };
+
+  buildInputs = [
+    coreutils
+    darwin.apple_sdk.frameworks.Cocoa
+    xcbuild
```

Comment:
I think `xcbuild` goes to `nativeBuildInputs`. Try setting `strictDeps = true;` and see if this is the case.

Reply:
You were correct, when I had `strictDeps = true;` and left `xcbuild` in `buildInputs` it failed -- but it worked when I moved `xcbuild` to `nativeBuildInputs`.

Resolved on latest commit!

Reply:
(I haven't been able to find super clear definitions differentiating between `buildInputs` and `nativeBuildInputs` -- would love any pointers to documentation that clearly describe them if possible so I can better understand)

Reply:
> (I haven't been able to find super clear definitions differentiating between `buildInputs` and `nativeBuildInputs` -- would love any pointers to documentation that clearly describe them if possible so I can better understand)

- `buildInputs` are for runtime dependencies, it's going to be included in the final package, because the runtime requires it.
- `nativeBuildInputs` are for build time dependencies, usually tools that aren't needs in the final package but are used during build.

This distinction is important when doing cross-compilation. The dependencies (usually tools) used for building are going to run in the `host architecture`, while dependencies included in the output package (which runtime/buildInputs is part of) will be in the `target architecture`.

So the question you should be asking is:
* Is this dependency needed for the application to run (runtime)?
  - If yes, `buildInputs`.
  - If no, `nativeBuildInputs`.

If it is build tools, like `cmake`, it's `nativeBuildInputs`.